packages/sanity/src/structure/panes/documentList/sheetList/batchPublish.tsx

```diff
@@ -0,0 +1,68 @@
+import {type ObjectSchemaType, type SanityClient, type SanityDocument} from 'sanity'
+
+type DisabledReason = 'LIVE_EDIT_ENABLED' | 'ALREADY_PUBLISHED' | 'NO_CHANGES'
+
+interface Validation {
+  id: string
+  reason: DisabledReason
+}
+
+export const batchPublish = ({
```

Comment:
This takes a list of documents and publish all of them in one transaction, if one fails all fail, ensuring all documents are published. 
Based on https://github.com/sanity-io/sanity/blob/next/packages/sanity/src/core/store/_legacy/document/document-pair/serverOperations/publish.ts#L6-L7